pkg/api/customization/logging/action.go

```diff
@@ -38,12 +38,19 @@ var (
 	fluentdCommand      = "fluentd -q --dry-run -i "
 	cleanCertDirCommand = "rm -rf "
 	tmpCertDirPrefix    = "/fluentd/etc/config/tmpCert"
+	fluentdLabelConfig  = `
+<label @FLUENT_LOG>
+    <match fluent.*>
+      	@type null
+    </match>
+</label>`
```

Comment:
Not very familiar with fluent, but wouldn't this get updated in [`MatchTemplate`](https://github.com/rancher/rancher/blob/master/pkg/controllers/user/logging/generator/templatematch.go#L5), why instead append to `dryRunConfigBuf` ?

Reply:
Dry run will deploy fluentd-tester which use the default fluentd configure here. https://github.com/rancher/fluentd/blob/master/package/fluent.conf#L4 , now fluentd will return warning when we use <match fluent.*>, so append fluentd label configure to avoid this warning. Because new release rancher may with old fluentd-tester deployed, so update the configure in rancher.

Related fluentd doc: 
https://docs.fluentd.org/deployment/logging#capture-fluentd-logs

Reply:
Can we add it in fluent.conf also, leaving both ? If so would be good to leave comments around reasoning for having both.

Reply:
Sure, add a comment [here](https://github.com/rancher/rancher/pull/27680/files#diff-5cc90d611c97b997b67fbc56fe215abdR41) and create a fluentd pr https://github.com/rancher/fluentd/pull/30

Reply:
lgtm now, adding jacob as reviewer to second PR